core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -799,6 +834,13 @@ trait EntityComponent {
         EntityRecordRawSqlQuery.batchHide(workspaceContext.workspaceIdAsUUID, entRefs).map(res => res.sum)
     }
 
+    // "deletes" entities of a certain type by hiding and renaming them
+    def hideType(workspaceContext: Workspace, entityType: String): ReadWriteAction[Int] = {
+      workspaceQuery.updateLastModified(workspaceContext.workspaceIdAsUUID) andThen
+        EntityAndAttributesRawSqlQuery.batchHideForType(workspaceContext, entityType) andThen
+        EntityRecordRawSqlQuery.batchHideType(workspaceContext.workspaceIdAsUUID, entityType).map(res => res.sum)
+    }
```

Comment:
I recognize you are following the pattern here - there are lots of other places where we update the workspace's lastModified and then perform other writes. Thinking about this pattern - I suspect it would be best to update the workspace's lastModified as the final step, after all other processing has completed.

1. In general, the other sql actions are more likely to have errors and so we should try them first; if they fail we wouldn't even attempt the lastModified update and there's one less thing to roll back
2. In general, the other sql actions will be bigger, so we shouldn't hold the lock on WORKSPACE while the other actions run; we should grab the WORKSPACE lock as the last thing

What's your take on this?